### PR TITLE
add patch for AlphaFold v2.2.2 to fix NaN problem with jax 0.3.9

### PR DESCRIPTION
**Pull request by ThomasHoffmann77**
(created using `eb --new-pr`)

edit (by @boegel): patch based on https://github.com/deepmind/alphafold/issues/513

### CODE REVIEW

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**SUCCESS**
Build succeeded for 3 out of 3 (2 easyconfigs in total)
gpu02.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz (skylake_avx512), 1 x NVIDIA Tesla V100-SXM2-32GB, 510.47.03, Python 3.6.8
See https://gist.github.com/b5f5bf7acd938ae761adcf5f67708911 for a full test report.

File: easybuild/easyconfigs/a/AlphaFold/AlphaFold-2.2.2-foss-2021a.eb [hunk 1]
```diff
@@ -87,6 +88,7 @@ components = [
             'dfda4dd5f9aba19fe2b6eb9a0ec583d12dcefdfee8ab8803fc57ad48d582db04',  # AlphaFold-2.0.0_n-cpu.patch
             '5363d403baf5ab73f4d3ddd72e19af9ff832de4b1d7ba25a5fbcc5846c1c890f',  # AlphaFold-2.1.0_fix-scp-path.patch
             '1a2e4e843bd9a4d15ee39e6c37cc63ba281311cc7a0a5610f0e43b52ef93faac',  # AlphaFold-2.0.1_setup_rm_tfcpu.patch
+            'fb00a3c9cb00fe87eef77d48dfd8e44c40ad8b820480c79c5937377da27fea5f',  # AlphaFold-2.2.2_jax038_NaN.patch
```
Comment (jfgrimm) [inline easybuild/easyconfigs/a/AlphaFold/AlphaFold-2.2.2-foss-2021a.eb:91]:
```suggestion
            '0b040ba99208cd9ad79b7d17db5720f7a447a019ac0ca786b3b019a55d1544e0',  # AlphaFold-2.2.2_jax038_NaN.patch
```

File: easybuild/easyconfigs/a/AlphaFold/AlphaFold-2.2.2-foss-2021a-CUDA-11.3.1.eb [hunk 2]
```diff
@@ -92,6 +93,7 @@ components = [
             'dfda4dd5f9aba19fe2b6eb9a0ec583d12dcefdfee8ab8803fc57ad48d582db04',  # AlphaFold-2.0.0_n-cpu.patch
             '5363d403baf5ab73f4d3ddd72e19af9ff832de4b1d7ba25a5fbcc5846c1c890f',  # AlphaFold-2.1.0_fix-scp-path.patch
             '1a2e4e843bd9a4d15ee39e6c37cc63ba281311cc7a0a5610f0e43b52ef93faac',  # AlphaFold-2.0.1_setup_rm_tfcpu.patch
+            'fb00a3c9cb00fe87eef77d48dfd8e44c40ad8b820480c79c5937377da27fea5f',  # AlphaFold-2.2.2_jax038_NaN.patch
```
Comment (jfgrimm) [inline easybuild/easyconfigs/a/AlphaFold/AlphaFold-2.2.2-foss-2021a-CUDA-11.3.1.eb:96]:
```suggestion
            '0b040ba99208cd9ad79b7d17db5720f7a447a019ac0ca786b3b019a55d1544e0',  # AlphaFold-2.2.2_jax038_NaN.patch
```

Comment (boegel) [reply easybuild/easyconfigs/a/AlphaFold/AlphaFold-2.2.2-foss-2021a-CUDA-11.3.1.eb]: Oh, I overlooked that `--inject-checksums` doesn't handle component checksums... 🤦 

Fixed in fc8be7736b (which was force pushed into the PR)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3307.joltik.os - Linux RHEL 8.4, x86_64, Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz (cascadelake), 1 x NVIDIA Tesla V100-SXM2-32GB, 510.73.08, Python 3.6.8
See https://gist.github.com/0c9644652b6d998d242151332723590d for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3904.accelgor.os - Linux RHEL 8.4, x86_64, AMD EPYC 7413 24-Core Processor (zen3), 1 x NVIDIA NVIDIA A100-SXM4-80GB, 510.73.08, Python 3.6.8
See https://gist.github.com/116e89b284db226dd37e201d7837d91f for a full test report.

**Comment by boegel** [comment]
Going in, thanks @ThomasHoffmann77!

**Pull request merged**